### PR TITLE
document SASL authentication and SSL encryption support

### LINKED ISSUE #382: Document the support of schema registry
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/6289

### Which part(s) of the docs might be affected or should be updated? And how?

Create source - Kafka -
We also need an integration guide that talks about these configurations. Not just listing them as parameters. 

### Reference

_No response_

### PR DESCRIPTION
**Pull request by hengm3467**
## Info
- **Description**: 
Update the Kafka/Redpanda topic to describe SSL and SASL configuration. Also updated the schema location & schema registry content and improved clarity/structure.

- **Related code PR**: 
[ Provide the link to the code PR here. For example, https://github.com/risingwavelabs/risingwave/pull/4085. Delete this part if there's no code PR related. ]

- **Related doc issue**: 
Resolves #382 , #381 

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

File: docs/sql/create-source/create-source-kafka-redpanda.md [hunk 1]
```diff
@@ -100,4 +104,282 @@ ROW SCHEMA LOCATION 'https://demo_bucket_name.s3-us-west-2.amazonaws.com/demo.pr
...
+
+## Read schemas from Schema Registry
+
+Confluent Schema Registry provides a serving layer for your metadata. It provides a RESTful interface for storing and retrieving your schemas.
+
+RisingWave supports reading schemas from a Schema Registry. The latest schema will be retrieved from the specified Schema Registry using the `TopicNameStrategy` strategy at the time the `CREATE SOURCE` statement is issued. Then the schema parser in RisingWave will automatically determine the columns and data types to use in the source.
+
+### Source schema change
+
+As long as the writer schema changes in a compatible way, RisingWave will continue using the original reader schema definition by mapping values from the new to the old schema version. To use a newer version of the writer schema in RisingWave, you need to drop and recreate the source.
```
Comment (tabVersion) [inline docs/sql/create-source/create-source-kafka-redpanda.md]: may take a look at https://materialize.com/docs/sql/create-source/#schema-evolution-1 . I don't know whether it is proper to mention reader and writer schema here

File: docs/sql/create-source/create-source-kafka-redpanda.md [hunk 2]
```diff
@@ -30,15 +33,16 @@ ROW FORMAT AVRO | JSON | PROTOBUF MESSAGE 'main_message'
...
 
-Specify the format of the stream in the `ROW FORMAT` section of your statement.
+### `ROW FORMAT` parameters
+
+|Field | Notes|
+|---|----|
+|data_format| Data format. Supported formats: `JSON`, `AVRO`, `PROTOBUF`|
+|message | Message for the format. Required for Avro and Protobuf.|
+|schema_location| Location of the schema file. It can be a local location, or a Web location that is in `s3://...` or `https://...` format. For Avro and Protobuf data, you must specify either a schema location or a schema registry but not both.|
+|schema_registry_url| Confluent Schema Registry URL. Example: `http://127.0.0.1:8081`. For Avro or Protobuf data, you must specify either a schema location or a schema registry but not both.|
```
Comment (tabVersion) [inline docs/sql/create-source/create-source-kafka-redpanda.md:44]: IIRC, there is another requirement on pb schema registry, cc @waruto210

**Review by tabVersion — APPROVED** [review]
generally LGTM

the authentication part is basically correct according to rdkafka doc. But I don't know about the encryption part, may need further check

File: docs/sql/create-source/create-source-kafka-redpanda.md (hunk 1 above)
Comment (hengm3467) [reply docs/sql/create-source/create-source-kafka-redpanda.md]: I just realized that this description is actually from the schema evolution section for Avro. Let me rephrase the paragraph.

File: docs/sql/create-source/create-source-kafka-redpanda.md [hunk 3]
```diff
@@ -100,4 +104,284 @@ ROW SCHEMA LOCATION 'https://demo_bucket_name.s3-us-west-2.amazonaws.com/demo.pr
...
+
+|Parameter| Notes|
+|---|---|
+|properties.security.protocol| For SASL/OAUTHBEARER without SSL, set to `SASL_PLAINTEXT`. For SASL/OAUTHBEARER with SSL, set to `SASL_SSL`.|
+|properties.sasl.mechanism|Set to `OAUTHBEARER`.|
+|properties.sasl.oauthbearer.config| |
+
+:::note
+
+For the definitions of the parameters, see the [librdkafka properties list](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md). Note that the parameters in the list assumes all parameters start with `properties.` and therefore do not include this prefix.
```
Comment (xx01cyx) [inline docs/sql/create-source/create-source-kafka-redpanda.md]: Maybe we should mention the limitations of OAUTHBEARER here as well. Since librdkafka provides a bunch of oauthbearer-related parameters (just search for "oauthbearer" on [this page](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md)), we don't want our users to get confused on the fact that they can only specify `properties.sasl.oauthbearer.config` in RisingWave.

File: docs/sql/create-source/create-source-kafka-redpanda.md [hunk 4]
```diff
@@ -100,4 +104,284 @@ ROW SCHEMA LOCATION 'https://demo_bucket_name.s3-us-west-2.amazonaws.com/demo.pr
...
+Simple Authentication and Security Layer (SASL) is a framework for authentication and data security in Internet protocols.
+
+RisingWave supports four SASL authentication mechanisms:
+
+- SASL/PLAIN
+- SASL/SCRAM
+- SASL/GSSAPI
+- SASL/OAUTHBEARER
+
+SSL encryption can be used concurrently with SASL authentication mechanisms. Due to a limitation, RisingWave does not support using SSL with SASL/OAUTHBEARER but supports using SSL with the other three SASL authentication mechanisms.
```
Comment (xx01cyx) [inline docs/sql/create-source/create-source-kafka-redpanda.md]: > SASL/OAUTHBEARER is only supported in unsecured mode, and OIDC is not allowed as well. This is because [rust-rdkafka](https://github.com/fede1024/rust-rdkafka) currently does not have a full support for OAuth based on [librdkafka](https://github.com/edenhill/librdkafka).

The `unsecured mode` here does not mean "w/o SSL", but means "[Unsecured Client Side Token](https://docs.confluent.io/platform/current/kafka/authentication_sasl/authentication_sasl_oauth.html#unsecured-client-side-token-creation-options-for-sasl-oauthbearer)" (Note that this CANNOT be used in production). So we actually support OAUTHBEARER w/ SSL; it is  OAUTHBEARER itself that is not fully supported.

File: docs/sql/create-source/create-source-kafka-redpanda.md (hunk 3 above)
Comment (xx01cyx) [reply docs/sql/create-source/create-source-kafka-redpanda.md]: You might first read the comment below and then read this comment. 🤣

File: docs/sql/create-source/create-source-kafka-redpanda.md (hunk 2 above)
Comment (hengm3467) [reply docs/sql/create-source/create-source-kafka-redpanda.md:44]: @waruto210 Could you please describe the requirement for pb schema registry? Thanks.

**Review by xx01cyx — APPROVED** [review]

File: docs/sql/create-source/create-source-kafka-redpanda.md (hunk 3 above)
Comment (hengm3467) [reply docs/sql/create-source/create-source-kafka-redpanda.md]: Thanks!

File: docs/sql/create-source/create-source-kafka-redpanda.md (hunk 2 above)
Comment (hengm3467) [reply docs/sql/create-source/create-source-kafka-redpanda.md:44]: I'll merge this PR for now, and add the requirement for pb schema registry later.

Comment (waruto210) [reply docs/sql/create-source/create-source-kafka-redpanda.md:44]: For protobuf, if user specify a schema location, the schema file must be a `FieldDescriptor`, which could be compiled from a `.proto` file:
```shell
protoc -I=$include_path --include_imports --descriptor_set_out=schema.pb schema.proto
```

Comment (waruto210) [reply docs/sql/create-source/create-source-kafka-redpanda.md:44]: And, the syntax of using schema registry is:
```sql
row schema location confluent schema registry
```
